type conversion for definition functions

```c
void s(char opt[]) {
    int l = strlen(opt) + 12;
    (void)(l);
    return;
}
int main(){
    return 0;
}
```
```go
package main

import "github.com/elliotchance/c2go/noarch"

func s(opt []byte) {
        var l int = noarch.Strlen(opt) + uint32(12)
        _ = (l)
        return
}

func main() {
        __init()
        return
}
func __init() {
}
```
```
○ → go run 1.go
# command-line-arguments
./1.go:8:33: invalid operation: noarch.Strlen(opt) + uint32(12) (mismatched types int and uint32)
```
AST:
```
|   | `-VarDecl 0x360b148 <col:5, col:27> col:9 used l 'int' cinit
|   |   `-ImplicitCastExpr 0x360b800 <col:13, col:27> 'int' <IntegralCast>
|   |     `-BinaryOperator 0x360b7d8 <col:13, col:27> 'unsigned long' '+'
|   |       |-CallExpr 0x360b740 <col:13, col:23> 'unsigned long'
|   |       | |-ImplicitCastExpr 0x360b728 <col:13> 'unsigned long (*)(const char *)' <FunctionToPointerDecay>
|   |       | | `-DeclRefExpr 0x360b690 <col:13> 'unsigned long (const char *)' Function 0x360b590 'strlen' 'unsigned long (const char *)'
|   |       | `-ImplicitCastExpr 0x360b788 <col:20> 'const char *' <BitCast>
|   |       |   `-ImplicitCastExpr 0x360b770 <col:20> 'char *':'char *' <LValueToRValue>
|   |       |     `-DeclRefExpr 0x360b6b8 <col:20> 'char *':'char *' lvalue ParmVar 0x35b87a0 'opt' 'char *':'char *'
|   |       `-ImplicitCastExpr 0x360b7c0 <col:27> 'unsigned long' <IntegralCast>
|   |         `-IntegerLiteral 0x360b7a0 <col:27> 'int' 12
```

--- a/tests/operators.c
+++ b/tests/operators.c
@@ -6,9 +6,29 @@
 
 void empty(){;}
 
+int sAdd(char *opt) {
+    int l = strlen(opt) + 12;
+    return l;
+}
+
+int sMul(char *opt) {
+    int l = strlen(opt) * 12;
+    return l;
+}
+
+int sMin(char *opt) {
+    int l = strlen(opt) - 12;
+    return l;
+}
+
+int sDiv(char *opt) {
+    int l = strlen(opt) / 12;
+    return l;
+}
+
 int main()
 {
-	plan(54);
+	plan(58);
 
     int i = 10;
     signed char j = 1;
@@ -215,5 +235,11 @@ int main()
 		case 1:	{ ia = 60; is_eq(ia,60); }
 	}
 
+	diag("Binary operators for definition function")
+	is_eq(sAdd("rrr"),15);
+	is_eq(sMul("rrr"),36);
+	is_eq(sMin("rrrrrrrrrrrrr"),1);
+	is_eq(sDiv("rrrrrrrrrrrr"),1);
+	
 	done_testing();
 }

--- a/transpiler/binary.go
+++ b/transpiler/binary.go
@@ -222,7 +222,7 @@ func transpileBinaryOperator(n *ast.BinaryOperator, p *program.Program, exprIsSt
 			leftType, preStmts, postStmts, nil
 	}
 
-	if operator == token.NEQ || operator == token.EQL || operator == token.LSS || operator == token.GTR || operator == token.AND {
+	if operator == token.NEQ || operator == token.EQL || operator == token.LSS || operator == token.GTR || operator == token.AND || operator == token.ADD || operator == token.SUB || operator == token.MUL || operator == token.QUO || operator == token.REM {
 		// We may have to cast the right side to the same type as the left
 		// side. This is a bit crude because we should make a better
 		// decision of which type to cast to instead of only using the type

